lib/Basics/ResourceUsage.cpp

```diff
@@ -22,40 +22,178 @@
 ////////////////////////////////////////////////////////////////////////////////
 
 #include "Basics/ResourceUsage.h"
+#include "Basics/Exceptions.h"
+#include "Basics/GlobalResourceMonitor.h"
+#include "Basics/debugging.h"
+#include "Basics/voc-errors.h"
 
 using namespace arangodb;
 
+ResourceMonitor::ResourceMonitor(GlobalResourceMonitor& global) noexcept
+    : _current(0),
+      _peak(0),
+      _limit(0),
+      _global(global) {}
+
 ResourceMonitor::~ResourceMonitor() {
   // this assertion is here to ensure that our memory usage tracking works
   // correctly, and everything that we accounted for is actually properly torn
   // down. the assertion will have no effect in production.
-  TRI_ASSERT(currentResources.memoryUsage.load(std::memory_order_relaxed) == 0);
+  TRI_ASSERT(_current.load(std::memory_order_relaxed) == 0);
```

Comment:
Yes, at the time the destructor is executed I would expect the necessary happens-before relation to be established, otherwise we would potentially have larger problems than observing an out-of-date value in `_current`